azurelinuxagent/common/protocol/extensions_goal_state.py

```diff
@@ -128,15 +157,17 @@ def __init__(self, incarnation, xml_text):
         if required_features_list is not None:
             self._parse_required_features(required_features_list)
 
-        self.status_upload_blob = findtext(xml_doc, "StatusUploadBlob")
+        self._status_upload_blob = findtext(xml_doc, "StatusUploadBlob")
         self.artifacts_profile_blob = findtext(xml_doc, "InVMArtifactsProfileBlob")
 
         status_upload_node = find(xml_doc, "StatusUploadBlob")
-        self.status_upload_blob_type = getattrib(status_upload_node, "statusBlobType")
-        logger.verbose("Extension config shows status blob type as [{0}]", self.status_upload_blob_type)
+        self._status_upload_blob_type = getattrib(status_upload_node, "statusBlobType")
+        logger.verbose("Extension config shows status blob type as [{0}]", self._status_upload_blob_type)
 
         self.in_vm_gs_metadata.parse_node(find(xml_doc, "InVMGoalStateMetaData"))
 
+        self._do_common_validations()
```

Comment:
Any reason for validating in 2 different places? (Line 133 and 169)?

Reply:
I want to be really, really sure the text is valid...

No, seriously, the second call is a leftover, will remove it. Thanks